pkg/common/config/config.go

```diff
@@ -22,46 +28,54 @@ const (
 
 // CONFIG conf
 var CONFIG goarchaius.ConfigurationFactory
+var once = sync.Once{}
 
-func init() {
-	// create go-archaius object
-	configFactory, err := goarchaius.NewConfigFactory(nil)
-	if err != nil {
-		fmt.Printf("failed to new config factory,  error: %+v", err)
-	}
-	CONFIG = configFactory
-	// init go-archaius
-	err = CONFIG.Init()
-	if err != nil {
-		fmt.Printf("failed to init config factory,  error: %+v", err)
-	}
-	//Add yaml files as config source
-	fSource := filesource.NewYamlConfigurationSource()
-	confLocation := GetConfigDirectory() + "/conf"
-	err = filepath.Walk(confLocation, func(location string, f os.FileInfo, err error) error {
-		if f == nil {
-			return err
+// config file  only support .yml or .yaml  !
+func InitializeConfig() {
+	once.Do(func() {
+		err := archaius.Init()
+		if err != nil {
+			log.LOGGER.Errorf("archaius init failed!")
```

Comment:
add a return here ?

Reply:
Maybe the function can have a error return type ?

Reply:
the same as above comments